pulsar-broker/src/main/java/org/apache/pulsar/broker/service/AbstractTopic.java

```diff
@@ -199,14 +220,34 @@ protected boolean isConsumersExceededOnTopic() {
         if (maxConsumersPerTopic > 0 && maxConsumersPerTopic <= getNumberOfConsumers()) {
             return true;
         }
+
+        final int maxSameAddressConsumers = brokerService.pulsar().getConfiguration()
+                .getMaxSameAddressConsumersPerTopic();
```

Comment:
I suggest to open a check independently and let the client know clearly that it is because the Consumer of a single IP reaches the upper limit.
Because we will add namespace and topic level Policies in the future. Should not be mixed with max_consumers_per_topic.
Producer is the same.